extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -370,22 +372,54 @@ export class Visibility {
 
     const visible = change.intersectionRatio * 100;
     for (let c = listeners.length - 1; c >= 0; c--) {
-      const shouldBeVisible = !!listeners[c]['shouldBeVisible'];
-      if (this.updateCounters_(visible, listeners[c], shouldBeVisible)
-          && this.viewer_.isVisible() == shouldBeVisible) {
-        this.prepareStateForCallback_(
-            listeners[c]['state'], change.boundingClientRect);
-        listeners[c].callback(listeners[c]['state']);
+      const listener = listeners[c];
+      const shouldBeVisible = !!listener.shouldBeVisible;
+      const config = listener.config;
+      const state = listener.state;
+      state[LAST_CHANGE_ENTRY] = change;
```

Comment:
Why do we use map access for something that looks very much like programmatic structure?